bluepysnap/circuit_validation.py

```diff
@@ -308,48 +313,81 @@ def _check_rotations():
+                    morph_dirs |= {(morph_path, extension)}
     if errors:
         return errors
-    _check_rotations()
-    errors += _check_files(
-        'morphology: {}[{}]'.format(group_name, group.file.filename),
-        (Path(components['morphologies_dir'], m + '.swc') for m in group_df['morphology']),
-        Error.WARNING)
-    bio_path = Path(components['biophysical_neuron_models_dir'])
+    for morph_path, extension in morph_dirs:
+        errors += _check_files(
+            'morphology: {}[{}]'.format(group_name, group.file.filename),
+            (Path(morph_path, m + '.' + extension) for m in group_df['morphology']),
+            Error.WARNING)
+    bio_path = Path(population['biophysical_neuron_models_dir'])
     errors += _check_files(
         'model_template: {}[{}]'.format(group_name, group.file.filename),
         (bio_path / _get_model_template_file(m) for m in group_df['model_template']),
         Error.WARNING)
     return errors
 
 
-def _check_nodes_group(group_df, group, config):
+def _check_nodes_group(group_df, group, config, population):
     """Validates nodes group in nodes population.
 
     Args:
         group_df (pd.DataFrame): nodes group in nodes .h5 file
         group (h5py.Group): nodes group in nodes .h5 file
         config (dict): resolved bluepysnap config
+        population (dict): definition of current population in config
 
     Returns:
         list: List of errors, empty if no errors
     """
+    errors = []
+    if 'type' in population and population['type'] not in NODE_TYPES:
+        errors.append(BbpError(Error.WARNING, 'Invalid node type: {}'.format(population['type'])))
     if 'model_type' not in group_df.columns:
-        return [fatal('Group {} of {} misses "model_type" field'
-                      .format(_get_group_name(group, parents=1), group.file.filename))]
+        return errors + [fatal('Group {} of {} misses "model_type" field'
+                               .format(_get_group_name(group, parents=1), group.file.filename))]
     if group_df['model_type'][0] == 'virtual':
-        return []
+        return errors
     if 'model_template' not in group_df.columns:
-        return [fatal('Group {} of {} misses "model_template" field'
-                      .format(_get_group_name(group, parents=1), group.file.filename))]
+        return errors + [fatal('Group {} of {} misses "model_template" field'
+                               .format(_get_group_name(group, parents=1), group.file.filename))]
     elif group_df['model_type'][0] == 'biophysical':
-        return _check_bio_nodes_group(group_df, group, config)
+        if 'components' not in config:
+            return errors + [fatal('No "components" in config')]
+        population = {**config['components'], **population}
+        return errors + _check_bio_nodes_group(group_df, group, population)
+    return errors
+
+
+def _check_populations_dict(populations_dict, populations_h5, file_name):
```

Comment:
Can we please use `populations_config` instead of `populations_dict`?

Reply:
Sure, no problem.